runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/HttpClientEngineConfig.kt

```diff
@@ -72,6 +73,8 @@ public open class HttpClientEngineConfig constructor(builder: Builder) {
      */
     public val proxySelector: ProxySelector = builder.proxySelector
 
+    public val minTlsVersion: TlsVersion = builder.minTlsVersion
```

Comment:
**Question**: Should we be forward looking and define something like `TlsContext` to group TLS settings?

Reply:
Some thoughts here--

I would prefer it from an organization perspective, but if we go that route I think we'd want to reconsider grouping all of the connection based things (timeouts, max conns) as well.

I think that would be liable to complicate operation config override for these settings (which AFAIK we wanted to revisit being able to do independently of setting the HTTP client). e.g. Instead of patching one field in the config, I now have to construct a new container, merge in what i want to stay the same, and apply overrides.

Not taking a hard stance here, more just trying to play out the different paths forward.

Reply:
`TlsContext` added and I've also moved the existing `alpn` setting into it alongside the new TLS minimum version.

Concerns about complications for overriding client config are most definitely valid but won't be solved in this PR. We'll tackle that in the near future with a larger overhaul.